### PR TITLE
Update migration case for network data transport

### PR DESCRIPTION
**Pull request by cliping**
1.. Update case:
  XXXX-294376 - VM live migration - network data transport - TCP

Signed-off-by: cliping <lcheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    variants test_case:
+        - tls:
+            virsh_migrate_extra = "--tls"
+            custom_pki_path = "/etc/pki/qemu"
+            qemu_tls = "yes"
+            server_cn = "ENTER.YOUR.EXAMPLE.SERVER_CN"
+            client_cn = "ENTER.YOUR.EXAMPLE.CLIENT_CN"
+            log_outputs = "/var/log/libvirt/libvirtd.log"
+            log_conf_dict = '{"log_level": "1", "log_outputs": "\"1:file:${log_outputs}\""}'
+            log_conf_type = "virtqemud"
```
Comment (Yingshun) [inline libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg:49]: I wonder if this case could check the log that's configured  by env_process?
The value of 'enable_libvirtd_debug_log' is 'yes' by default, which means libvirtd log will be set for each test by default.
Is it possible to reuse that log file on local host?

File: libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        :param params: Dictionary with the test parameter
+        :param test: test object
+        :return: log configure object
+        """
+        test.log.info("Setup for TLS transport.")
+        dest_uri = params.get("virsh_migrate_desturi")
+
+        log_conf_obj = update_local_and_remote_log_file(params, test)
+        migration_base.setup_conn_obj('tls', conn_obj_list, params, test)
+        migration_base.setup_conn_obj('tcp', conn_obj_list, params, test)
```
Comment (Yingshun) [inline libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py:61]: Do we need tcp connection for this case?

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 migration.network_data_transports.tcp
JOB ID     : 81bccec4d3bf613b32c811bb1a5a3cf776672994
JOB LOG    : /var/lib/avocado/job-results/job-2022-06-10T22.06-81bccec/job.log
 (1/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_precopy.p2p: STARTED
 (1/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_precopy.p2p: PASS (222.78 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_precopy.non_p2p: STARTED
 (2/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_precopy.non_p2p: PASS (221.94 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_postcopy.p2p: STARTED
 (3/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_postcopy.p2p: PASS (190.72 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_postcopy.non_p2p: STARTED
 (4/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tcp.with_postcopy.non_p2p: PASS (191.24 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-06-10T22.06-81bccec/results.html
JOB TIME   : 829.17 s

#avocado run --vt-type libvirt --vt-machine-type q35 migration.network_data_transports.tls
JOB ID     : 44912349700867dce90dd55e68c2f004cb5fb7ca
JOB LOG    : /var/lib/avocado/job-results/job-2022-06-11T12.23-4491234/job.log
 (1/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_precopy.p2p: STARTED
 (1/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_precopy.p2p: PASS (297.29 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_precopy.non_p2p: STARTED
 (2/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_precopy.non_p2p: PASS (296.41 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_postcopy.p2p: STARTED
 (3/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_postcopy.p2p: PASS (274.80 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_postcopy.non_p2p: STARTED
 (4/4) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tls.with_postcopy.non_p2p: PASS (276.31 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-06-11T12.23-4491234/results.html
JOB TIME   : 1147.32 s

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 migration.network_data_transports.tunnelled
JOB ID     : 5ee68673bbd162bcadef0f68bb40ceaba0500247
JOB LOG    : /var/lib/avocado/job-results/job-2022-06-12T00.01-5ee6867/job.log
 (1/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.tls: STARTED
 (1/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.tls: PASS (254.02 s)
 (2/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.tcp: STARTED
 (2/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.tcp: PASS (210.25 s)
 (3/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.ssh: STARTED
 (3/3) type_specific.io-github-autotest-libvirt.migration.network_data_transports.tunnelled.ssh: PASS (210.70 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-06-12T00.01-5ee6867/results.html
JOB TIME   : 676.97 s

File: libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg (hunk 1 above)
Comment (cliping) [reply libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg]: Updated, thanks.

File: libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py (hunk 2 above)
Comment (cliping) [reply libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py]: Yes, "qemu+tcp://${migrate_dest_host}/system" need tcp connection.

**Comment by cliping** [comment]
@fangge1212 , @Yingshun , @dzhengfy , please take a look, thanks.

File: provider/migration/migration_base.py [hunk 3]
```diff
@@ -260,3 +266,33 @@ def execute_statistics_command(params):
         virsh.domblkinfo(vm_name, disk_source, **debug_kargs)
         virsh.domstats(vm_name, **debug_kargs)
         virsh.dommemstat(vm_name, **debug_kargs)
+
+
+def check_network_connection(params):
```
Comment (Yingshun) [inline provider/migration/migration_base.py:271]: Can we reuse virttest.utls_libvirt.libvirt_network.check_established()?

File: provider/migration/base_steps.py [hunk 4]
```diff
@@ -1,102 +1,209 @@
...
+    def check_local_or_remote_log(self):
+        """
+        Check local or remote log file
+
+        """
+        check_str_local_log = self.params.get("check_str_local_log", "")
+        check_str_remote_log = self.params.get("check_str_remote_log", "")
+        log_file = self.params.get("libvirtd_debug_file")
+        if check_str_local_log:
+            libvirt.check_logfile(check_str_local_log, log_file)
```
Comment (dzhengfy) [inline provider/migration/base_steps.py:196]: Is str_in_log needed on local checking? I suggest we use str_in_log for local and remote checking or using different boolean variables for local and remote is better. Then we can merge codes below to only write libvirt.check_logfile() once.  What do you think of?

File: provider/migration/migration_base.py [hunk 5]
```diff
@@ -260,3 +266,33 @@ def execute_statistics_command(params):
...
+    server_ip = params.get("server_ip", params.get("remote_ip"))
+    expect_num = params.get("expect_num")
+    cmd = "netstat -tuxnap|grep -E \".*%s.*ESTABLISHED.*qemu-kvm\" | wc -l" % server_ip
+    cmd_result = remote.run_remote_cmd(cmd, params).stdout_text.strip()
+    if cmd_result != expect_num:
+        raise exceptions.TestFail("Exist network connection betweent src qemu "
+                                  "and target qemu: {}".format(cmd_result))
+
+
+def check_unix_socket(params):
```
Comment (dzhengfy) [inline provider/migration/migration_base.py:286]: Is this checking very common?

File: libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tcp.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        migration_obj.setup_default()
+
+    def cleanup_tcp():
+        """
+        Cleanup steps for TCP transport
+
+        """
+        test.log.info("Cleanup steps for TCP transport.")
+        migration_obj.cleanup_default()
+        migration_base.cleanup_conn_obj(conn_obj_list, test)
```
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tcp.py]: Updated, add setup_connectio()/cleanup_connection() in MigrationBase().

File: provider/migration/migration_base.py (hunk 3 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Move this part to virttest.utls_libvirt.libvirt_network.check_established().
https://github.com/avocado-framework/avocado-vt/pull/3433

File: libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        """
+        Setup for TLS transport
+
+        """
+        test.log.info("Setup for TLS transport.")
+
+        global remote_conf_obj
+        remote_conf_obj = update_remote_conf(params, test)
+        conn_obj_list.append(migration_base.setup_conn_obj('tls', params, test))
+        conn_obj_list.append(migration_base.setup_conn_obj('tcp', params, test))
```
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/src/migration/network_data_transports/migration_network_data_transport_tls.py]: Yes, need to setup tcp. And I moved tls case to another PR. Thanks.

File: libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            virsh_migrate_options = '--live --verbose'
+    variants:
+        - with_precopy:
+            migrate_speed = "10"
+        - with_postcopy:
+            stress_package = "stress"
+            stress_args = "--cpu 8 --io 4 --vm 2 --vm-bytes 128M --timeout 20s"
+            postcopy_options = '--postcopy --timeout 10 --timeout-postcopy'
+    variants test_case:
+        - tls:
```
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/cfg/migration/network_data_transports/migration_network_data_transport_tls.cfg]: ok, I will update this name in another PR.

File: provider/migration/migration_base.py (hunk 5 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Move this part to virttest.utls_libvirt.libvirt_network.check_established().
https://github.com/avocado-framework/avocado-vt/pull/3433

**Comment by cliping** [comment]
#avocado run --vt-type libvirt --vt-machine-type q35 migration.migration_uri.network_data_transport.tcp
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : bcbd297a3d10c8fb9380a8c8e6ea9c0e47125cb3
JOB LOG    : /var/lib/avocado/job-results/job-2022-06-26T12.02-bcbd297/job.log
 (1/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_precopy.p2p: STARTED
 (1/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_precopy.p2p: PASS (230.45 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_precopy.non_p2p: STARTED
 (2/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_precopy.non_p2p: PASS (230.15 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_postcopy.p2p: STARTED
 (3/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_postcopy.p2p: PASS (199.66 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_postcopy.non_p2p: STARTED
 (4/4) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.with_postcopy.non_p2p: PASS (199.34 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-06-26T12.02-bcbd297/results.html
JOB TIME   : 862.02 s

**Review by Yingshun — APPROVED** [review]
LGTM

File: provider/migration/base_steps.py (hunk 4 above)
Comment (cliping) [reply provider/migration/base_steps.py]: Updated, thanks.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
Depend on https://github.com/avocado-framework/avocado-vt/pull/3433

**Pull request merged**